### PR TITLE
Add Cat Avatar Generator Addon

### PR DESCRIPTION
**Pull request by fabrixxm**

### CODE REVIEW

**Comment by fabrixxm** [comment]
whoa! at first try? no comments on code style? :-)

**Comment by annando** [comment]
There was a reason, why I hadn't accepted it - I spotted these violations and thought you would let your virtual dogs chase me, when I would do it - and then I saw you accepting it without any complaint ;-)

File: catavatar/catavatar.php [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+}
+
+
+/**
+ * Returns the URL to the cat avatar
+ *
+ * @param $a array
+ * @param &$b array
+ */
+function catavatar_lookup($a, &$b) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Please use `(\Friendica\App $a, &$b)` parameter list instead.

File: catavatar/catavatar.php [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+
+/**
+ * Returns image for user id
+ *
+ * @throws NotFoundException
+ *
+ * @TODO: support sizes
+ */
+function catavatar_content($a) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Please use `(\Friendica\App $a)` parameter list instead.

File: catavatar/catavatar.php [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+		'$morecat' => L10n::t('More Random Cat!'),
+		'$emailcat' => L10n::t('Reset to email Cat'),
+		'$seed' => PConfig::get(local_user(), "catavatar", "seed", false),
+		'$header' => L10n::t('Cat Avatar').' '.L10n::t('Settings'),
+	]);
+	return;
+}
+
+function catavatar_addon_settings_post(&$a, &$s) {
+	if(! local_user())
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add braces to every conditional statement.

File: catavatar/catavatar.php [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+/**
+ * Returns image for user id
+ *
+ * @throws NotFoundException
+ *
+ * @TODO: support sizes
+ */
+function catavatar_content($a) {
+	if ($a->argc < 2 || $a->argc > 3)
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add braces to every conditional statement.

File: catavatar/catavatar.php [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+ * 
+ * This PHP is licensed under the short and simple permissive:
+ * [MIT License](https://en.wikipedia.org/wiki/MIT_License)
+ * 
+**/
+
+function build_cat($seed='', $size=0){
+
+	// init random seed
+	if($seed) srand( hexdec(substr(md5($seed),0,6)) );
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+		$dest = imagecreatetruecolor($size, $size);
+		imagealphablending($dest, false);
+		imagesavealpha($dest, true);
+		imagecopyresampled($dest, $cat, 0, 0, 0, 0, $size, $size, CATAVATAR_SIZE, CATAVATAR_SIZE);
+		imagedestroy($cat);
+		$cat = $dest;
+	}
+	
+	// restore random seed
+	if($seed) srand();
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap conditionals with braces.

File: catavatar/catavatar.php [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+	$seed = PConfig::get(local_user(), "catavatar", "seed", md5(trim(strtolower($user['email']))));
+	$imageurl = preg_replace('/[^A-Za-z0-9\._-]/', '', $seed); 
+	$imageurl = substr($imageurl,0,35).'';
+	$cachefile = get_cachefile($imageurl);
+	if ($cachefile != "" && file_exists($cachefile)) {
+		unlink($cachefile);
+	}
+		
+		
+	if (x($_POST,"catavatar-usecat")) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Obsolete: Please use `!empty()` instead of `x()`.

File: catavatar/catavatar.php [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+
+		Worker::add(PRIORITY_LOW, 'ProfileUpdate', local_user());
+		
+		info(L10n::t("Meow!"));
+		return;
+	}
+	
+
+
+	if (x($_POST,"catavatar-morecat")) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Obsolete: Please use `!empty()` instead of `x()`.

File: catavatar/catavatar.php [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+		PConfig::set(local_user(), "catavatar", "seed", time());
+	}
+
+	if (x($_POST,"catavatar-emailcat")) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Obsolete: Please use `!empty()` instead of `x()`.

File: catavatar/catavatar.php [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+	
+	logger("unregistered catavatar");
+}
+
+
+function catavatar_addon_settings(&$a, &$s) {
+	if(! local_user())
+		return;
+
+	$t = get_markup_template("settings.tpl", "addon/catavatar/" );
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+
+/**
+ * Installs the addon hook
+ */
+function catavatar_install() {
+	Addon::registerHook('avatar_lookup', 'addon/catavatar/catavatar.php', 'catavatar_lookup');
+	Addon::registerHook('addon_settings', 'addon/catavatar/catavatar.php', 'catavatar_addon_settings');
+	Addon::registerHook('addon_settings_post', 'addon/catavatar/catavatar.php', 'catavatar_addon_settings_post');
+
+	logger("registered catavatar");
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+ */
+use Friendica\Core\Addon;
+use Friendica\Core\Config;
+use Friendica\Core\L10n;
+use Friendica\Core\Worker;
+use Friendica\Core\PConfig;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Network\HTTPException\NotFoundException;
+
+define("CATAVATAR_SIZE", 256);
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 13]
```diff
@@ -0,0 +1,306 @@
...
+
+/**
+ * Removes the addon hook
+ */
+function catavatar_uninstall() {
+	Addon::unregisterHook('avatar_lookup', 'addon/catavatar/catavatar.php', 'catavatar_lookup');
+	Addon::unregisterHook('addon_settings', 'addon/catavatar/catavatar.php', 'catavatar_addon_settings');
+	Addon::unregisterHook('addon_settings_post', 'addon/catavatar/catavatar.php', 'catavatar_addon_settings_post');
+	
+	logger("unregistered catavatar");
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 14]
```diff
@@ -0,0 +1,306 @@
...
+}
+
+
+function catavatar_addon_settings(&$a, &$s) {
+	if(! local_user())
+		return;
+
+	$t = get_markup_template("settings.tpl", "addon/catavatar/" );
+	$s = replace_macros ($t, [
+		'$postpost' => x($_POST,"catavatar-morecat") || x($_POST,"catavatar-emailcat"),
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 15]
```diff
@@ -0,0 +1,306 @@
...
+		'$uncache' => time(),
+		'$uid' => local_user(),
+		'$usecat' => L10n::t('Use Cat as Avatar'),
+		'$morecat' => L10n::t('More Random Cat!'),
+		'$emailcat' => L10n::t('Reset to email Cat'),
+		'$seed' => PConfig::get(local_user(), "catavatar", "seed", false),
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+	// delete the current cached cat avatar
+	$user = dba::selectFirst('user', ['email'],
+		[
+			'uid' => $uid,
+			'blocked' => 0,
+			'account_expired' => 0,
+			'account_removed' => 0,
+		]
+	);
+	$seed = PConfig::get(local_user(), "catavatar", "seed", md5(trim(strtolower($user['email']))));
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+	$imageurl = preg_replace('/[^A-Za-z0-9\._-]/', '', $seed); 
+	$imageurl = substr($imageurl,0,35).'';
+	$cachefile = get_cachefile($imageurl);
+	if ($cachefile != "" && file_exists($cachefile)) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php (hunk 7 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 18]
```diff
@@ -0,0 +1,306 @@
...
+		unlink($cachefile);
+	}
+		
+		
+	if (x($_POST,"catavatar-usecat")) {
+		$url = $a->get_baseurl()."/catavatar/".local_user();
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 19]
```diff
@@ -0,0 +1,306 @@
...
+		
+		// set the catavatar url as avatar url in contact and default profile
+		// and set profile to 0 to current photo
+		// I'm not sure it's the correct way to do this...
+		$r = dba::update('contact', 
+			['photo'=>$url."/4", 'thumb'=>$url."/5", 'micro'=>$url."/6", 'avatar-date'=>DateTimeFormat::utcNow()],
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+			['photo'=>$url."/4", 'thumb'=>$url."/5", 'micro'=>$url."/6", 'avatar-date'=>DateTimeFormat::utcNow()], 
+			['uid'=>local_user(), 'self'=>1]
+		);
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+		$r = dba::update('profile', 
+			['photo'=>$url."/4", 'thumb'=>$url."/5"],
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 21]
```diff
@@ -0,0 +1,306 @@
...
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+
+		// Update global directory in background
+		$url = $a->get_baseurl() . '/profile/' . $a->user['nickname'];
+		if ($url && strlen(Config::get('system','directory'))) {
+			Worker::add(PRIORITY_LOW, "Directory", $url);
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 22]
```diff
@@ -0,0 +1,306 @@
...
+		}
+
+		Worker::add(PRIORITY_LOW, 'ProfileUpdate', local_user());
+		
+		info(L10n::t("Meow!"));
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php (hunk 8 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 23]
```diff
@@ -0,0 +1,306 @@
...
+		return;
+	}
+	
+
+
+	if (x($_POST,"catavatar-morecat")) {
+		PConfig::set(local_user(), "catavatar", "seed", time());
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php (hunk 9 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 24]
```diff
@@ -0,0 +1,306 @@
...
+	}
+
+	if (x($_POST,"catavatar-emailcat")) {
+		PConfig::delete(local_user(), "catavatar", "seed");
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 25]
```diff
@@ -0,0 +1,306 @@
...
+ * @param $a array
+ * @param &$b array
+ */
+function catavatar_lookup($a, &$b) {
+	$user = dba::selectFirst('user', ['uid'],['email'=>$b['email']]);
+	
+	$url = $a->get_baseurl().'/catavatar/'.$user['uid'];
+
+	switch($b['size']) {
+		case 175: $url.="/4"; break;
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 26]
```diff
@@ -0,0 +1,306 @@
...
+		case 80: $url.="/5"; break;
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 27]
```diff
@@ -0,0 +1,306 @@
...
+		case 47: $url.="/6"; break;
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 28]
```diff
@@ -0,0 +1,306 @@
...
+			'blocked' => 0,
+			'account_expired' => 0,
+			'account_removed' => 0,
+		]
+	);	
+	
+	if ($user === False)
+		throw new NotFoundException();
+	
+	$seed = PConfig::get(local_user(), "catavatar", "seed", md5(trim(strtolower($user['email']))));
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 29]
```diff
@@ -0,0 +1,306 @@
...
+	//echo "<pre>"; var_dump($hash); killme();
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please remove debug code.

File: catavatar/catavatar.php [hunk 30]
```diff
@@ -0,0 +1,306 @@
...
+
+	
+	// from cat-avatar-generator.php
+
+	$imageurl = $seed."-".$size;
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 31]
```diff
@@ -0,0 +1,306 @@
...
+	$imageurl = preg_replace('/[^A-Za-z0-9\._-]/', '', $imageurl); 
+	$imageurl = substr($imageurl,0,35).'';
+	$cachefile = get_cachefile($imageurl);
+	$cachetime = 604800; # 1 week (1 day = 86400)
+
+	// Serve from the cache if it is younger than $cachetime
+	if ($cachefile != "" && file_exists($cachefile) && time() - $cachetime < filemtime($cachefile)) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 32]
```diff
@@ -0,0 +1,306 @@
...
+	}
+
+	// ...Or start generation
+	ob_start(); 
+
+	// render the picture:
+	build_cat($seed, $size);
+
+	// Save/cache the output to a file
+	if ($cachefile!=""){
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 33]
```diff
@@ -0,0 +1,306 @@
...
+		'body' => rand(1,15),
+		'fur' => rand(1,10),
+		'eyes' => rand(1,15),
+		'mouth' => rand(1,10),
+		'accessorie' => rand(1,20)
+	);
+
+	// create backgound
+	$cat = @imagecreatetruecolor(CATAVATAR_SIZE, CATAVATAR_SIZE)
+		or die("GD image create failed");
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please use single-quotes or double-quotes but not a mix and match.

File: catavatar/catavatar.php [hunk 34]
```diff
@@ -0,0 +1,306 @@
...
+		[
+			'uid' => $uid,
+			'blocked' => 0,
+			'account_expired' => 0,
+			'account_removed' => 0,
+		]
+	);
+	$seed = PConfig::get(local_user(), "catavatar", "seed", md5(trim(strtolower($user['email']))));
+	$imageurl = preg_replace('/[^A-Za-z0-9\._-]/', '', $seed); 
+	$imageurl = substr($imageurl,0,35).'';
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php (hunk 7 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 35]
```diff
@@ -0,0 +1,306 @@
...
+		);
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+
+		// Update global directory in background
+		$url = $a->get_baseurl() . '/profile/' . $a->user['nickname'];
+		if ($url && strlen(Config::get('system','directory'))) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php (hunk 8 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php (hunk 9 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 36]
```diff
@@ -0,0 +1,306 @@
...
+
+
+/**
+ * Returns the URL to the cat avatar
+ *
+ * @param $a array
+ * @param &$b array
+ */
+function catavatar_lookup($a, &$b) {
+	$user = dba::selectFirst('user', ['uid'],['email'=>$b['email']]);
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 37]
```diff
@@ -0,0 +1,306 @@
...
+	
+	$seed = PConfig::get(local_user(), "catavatar", "seed", md5(trim(strtolower($user['email']))));
+	//echo "<pre>"; var_dump($hash); killme();
+
+	
+	// from cat-avatar-generator.php
+
+	$imageurl = $seed."-".$size;
+	$imageurl = preg_replace('/[^A-Za-z0-9\._-]/', '', $imageurl); 
+	$imageurl = substr($imageurl,0,35).'';
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 38]
```diff
@@ -0,0 +1,306 @@
...
+**/
+
+function build_cat($seed='', $size=0){
+
+	// init random seed
+	if($seed) srand( hexdec(substr(md5($seed),0,6)) );
+
+	// throw the dice for body parts
+	$parts = array(
+		'body' => rand(1,15),
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 39]
```diff
@@ -0,0 +1,306 @@
...
+		'fur' => rand(1,10),
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 40]
```diff
@@ -0,0 +1,306 @@
...
+		'eyes' => rand(1,15),
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 41]
```diff
@@ -0,0 +1,306 @@
...
+		'mouth' => rand(1,10),
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 42]
```diff
@@ -0,0 +1,306 @@
...
+		'accessorie' => rand(1,20)
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 43]
```diff
@@ -0,0 +1,306 @@
...
+	);
+
+	// create backgound
+	$cat = @imagecreatetruecolor(CATAVATAR_SIZE, CATAVATAR_SIZE)
+		or die("GD image create failed");
+	$white = imagecolorallocate($cat, 255, 255, 255);
+	imagefill($cat,0,0,$white);
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php [hunk 44]
```diff
@@ -0,0 +1,306 @@
...
+
+	// add parts
+	foreach($parts as $part => $num){
+		$file = dirname(__FILE__).'/avatars/'.$part.'_'.$num.'.png';
+
+		$im = @imagecreatefrompng($file);
+		if(!$im) die('Failed to load '.$file);
+		imageSaveAlpha($im, true);
+		imagecopy($cat,$im,0,0,0,0,CATAVATAR_SIZE,CATAVATAR_SIZE);
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space after commas.

File: catavatar/catavatar.php (hunk 18 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 19 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 45]
```diff
@@ -0,0 +1,306 @@
...
+		$url = $a->get_baseurl()."/catavatar/".local_user();
+		
+		// set the catavatar url as avatar url in contact and default profile
+		// and set profile to 0 to current photo
+		// I'm not sure it's the correct way to do this...
+		$r = dba::update('contact', 
+			['photo'=>$url."/4", 'thumb'=>$url."/5", 'micro'=>$url."/6", 'avatar-date'=>DateTimeFormat::utcNow()], 
+			['uid'=>local_user(), 'self'=>1]
+		);
+		if ($r===false) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 46]
```diff
@@ -0,0 +1,306 @@
...
+		
+	if (x($_POST,"catavatar-usecat")) {
+		$url = $a->get_baseurl()."/catavatar/".local_user();
+		
+		// set the catavatar url as avatar url in contact and default profile
+		// and set profile to 0 to current photo
+		// I'm not sure it's the correct way to do this...
+		$r = dba::update('contact', 
+			['photo'=>$url."/4", 'thumb'=>$url."/5", 'micro'=>$url."/6", 'avatar-date'=>DateTimeFormat::utcNow()], 
+			['uid'=>local_user(), 'self'=>1]
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 20 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 47]
```diff
@@ -0,0 +1,306 @@
...
+		);
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+		$r = dba::update('profile', 
+			['photo'=>$url."/4", 'thumb'=>$url."/5"], 
+			['uid'=>local_user(), 'is-default'=>1]
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 48]
```diff
@@ -0,0 +1,306 @@
...
+		);
+		if ($r===false) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 49]
```diff
@@ -0,0 +1,306 @@
...
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+		$r = dba::update('photo', 
+			['profile'=>0],
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 50]
```diff
@@ -0,0 +1,306 @@
...
+			['uid'=>local_user(), 'is-default'=>1]
+		);
+		if ($r===false) {
+			notice(L10n::t('There was an error, the cat ran away.'));
+			return;
+		}
+
+		$r = dba::update('photo', 
+			['profile'=>0], 
+			['uid'=>local_user(), 'profile'=>1]
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 51]
```diff
@@ -0,0 +1,306 @@
...
+		);
+		if ($r===false) {
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 36 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 52]
```diff
@@ -0,0 +1,306 @@
...
+/**
+ * Returns the URL to the cat avatar
+ *
+ * @param $a array
+ * @param &$b array
+ */
+function catavatar_lookup($a, &$b) {
+	$user = dba::selectFirst('user', ['uid'],['email'=>$b['email']]);
+	
+	$url = $a->get_baseurl().'/catavatar/'.$user['uid'];
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 25 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 27 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 26 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 53]
```diff
@@ -0,0 +1,306 @@
...
+	$user = dba::selectFirst('user', ['email'],
+		[
+			'uid' => $uid,
+			'blocked' => 0,
+			'account_expired' => 0,
+			'account_removed' => 0,
+		]
+	);	
+	
+	if ($user === False)
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap conditionals with braces.

File: catavatar/catavatar.php (hunk 30 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 37 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 54]
```diff
@@ -0,0 +1,306 @@
...
+	$imageurl = preg_replace('/[^A-Za-z0-9\._-]/', '', $imageurl); 
+	$imageurl = substr($imageurl,0,35).'';
+	$cachefile = get_cachefile($imageurl);
+	$cachetime = 604800; # 1 week (1 day = 86400)
+
+	// Serve from the cache if it is younger than $cachetime
+	if ($cachefile != "" && file_exists($cachefile) && time() - $cachetime < filemtime($cachefile)) {
+		header('Pragma: public');
+		header('Cache-Control: max-age=86400');
+		header('Expires: '. gmdate('D, d M Y H:i:s \G\M\T', time() + 86400));
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 32 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 5 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap conditionals with braces.

Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please remove extra spaces in `srand()` argument list.

File: catavatar/catavatar.php [hunk 55]
```diff
@@ -0,0 +1,306 @@
...
+
+	// create backgound
+	$cat = @imagecreatetruecolor(CATAVATAR_SIZE, CATAVATAR_SIZE)
+		or die("GD image create failed");
+	$white = imagecolorallocate($cat, 255, 255, 255);
+	imagefill($cat,0,0,$white);
+
+	// add parts
+	foreach($parts as $part => $num){
+		$file = dirname(__FILE__).'/avatars/'.$part.'_'.$num.'.png';
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php [hunk 56]
```diff
@@ -0,0 +1,306 @@
...
+
+		$im = @imagecreatefrompng($file);
+		if(!$im) die('Failed to load '.$file);
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap conditionals with braces.

File: catavatar/catavatar.php [hunk 57]
```diff
@@ -0,0 +1,306 @@
...
+		imagedestroy($cat);
+		$cat = $dest;
+	}
+	
+	// restore random seed
+	if($seed) srand();
+
+	header('Pragma: public');
+	header('Cache-Control: max-age=86400');
+	header('Expires: '. gmdate('D, d M Y H:i:s \G\M\T', time() + 86400));
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please wrap operators with spaces.

File: catavatar/catavatar.php (hunk 32 above)
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space before conditional opening brace.

File: catavatar/catavatar.php [hunk 58]
```diff
@@ -0,0 +1,306 @@
...
+	);
+
+	// create backgound
+	$cat = @imagecreatetruecolor(CATAVATAR_SIZE, CATAVATAR_SIZE)
+		or die("GD image create failed");
+	$white = imagecolorallocate($cat, 255, 255, 255);
+	imagefill($cat,0,0,$white);
+
+	// add parts
+	foreach($parts as $part => $num){
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space before opening braces.

File: catavatar/catavatar.php [hunk 59]
```diff
@@ -0,0 +1,306 @@
...
+		case 80: $url.="/5"; break;
+		case 47: $url.="/6"; break;
+	}
+
+	$b['url'] = $url;
+	$b['success'] = true;
+}
+
+
+function catavatar_module(){}
```
Comment (MrPetovan) [inline catavatar/catavatar.php]: Standards: Please add a space before opening braces.

Comment (fabrixxm) [reply catavatar/catavatar.php]: as is a empty function, what's the standard here?

```php
function catavatar_module()
{}

// or

function catavatar_module()
{
}
```

Comment (MrPetovan) [reply catavatar/catavatar.php]: I would go with `function catavatar_module() {}` as it just is a (wrong) way of indicating the addon provides a module.

Comment (fabrixxm) [reply catavatar/catavatar.php]: Yeah. Ok.
( in a future hopefully we will see something like
```php
class CatAvatar extends Addon implements Module {
```
:-D )

Comment (MrPetovan) [reply catavatar/catavatar.php]: I'd rather see
```php
class CatAvatar extends Addon
{
    public function __construct(\Friendica\App $a)
    {
        $a->addRoute('catavatar', '\Friendica\Addon\Module\CatAvatar');
        ...
    }
    ...
}
```
and have the module in a separate file in the addon.

**Pull request merged**